qutebrowser/misc/sessions.py

```diff
@@ -459,6 +464,34 @@ def list_sessions(self):
                 sessions.append(base)
         return sorted(sessions)
 
+    def add_url_to_session(self, name):
+        """Appends the current tab's URL to a named session."""
+        data = self._load_session_file(name)
+
+        if data is None:
+            raise SessionError("Got empty session file")
+
+        winlist = objreg.window_registry
+        current_tab = None
+        for win_id in sorted(winlist):
+            tabbed_browser = objreg.get('tabbed-browser', scope='window',
+                                        window=win_id)
+
+            active_window = QApplication.instance().activeWindow()
+            if getattr(active_window, 'win_id', None) == win_id:
+                current_tab = tabbed_browser.widget.currentWidget()
+
+        data['windows'][0]['tabs'].append(self._save_tab(current_tab, False))
```

Comment:
If there is a malformed session file, I think this will cause a crash. I'm not sure if there's a clean way to chain dict lookups safely in python though...

Ideally people could add a URL to an empty file as well, and have it generate a valid session with one tab. Could you add a test for that? :)

Also, is `[0]` right here? I don't know if there's a "better" way to find out which window to add the tab to.

Reply:
> If there is a malformed session file, I think this will cause a crash. I'm not sure if there's a clean way to chain dict lookups safely in python though...

It'll have to be something like this:
```
windows = data.get('windows', [])
if not windows:
    return / or raise
tabs = windows[0].get('tabs', [])
```
I'll make some changes to make this safer.

> Ideally people could add a URL to an empty file as well, and have it generate a valid session with one tab. Could you add a test for that? 

I thought about this and i guess the prompt will require an existing session. But i'll double check this as well and add a test.

> Also, is [0] right here? I don't know if there's a "better" way to find out which window to add the tab to.

Yea this is a problem. If you have a session with multiple windows, how can i know on which window the new URL will be inserted into? the obvious solutions would be either the first one or the last one. I don't think i can "chain" prompts in qutebrowser, as in: ask for the session selection first and then list windows and ask for the window selection... or can i?

Reply:
It seems fine to me to pick the first one, we could ask for window selection using something like buffer but I don't think that's worth the complexity.